unstructured/ingest/v2/processes/connectors/chroma.py

```diff
@@ -0,0 +1,284 @@
+        """
+        Prepares dictionary in the format that Chroma requires
+        """
+        element_id = data.get("element_id", str(uuid.uuid4()))
+        return {
+            "id": element_id,
+            "embedding": data.pop("embeddings", None),
+            "document": data.pop("text", None),
+            "metadata": flatten_dict(data, separator="-", flatten_lists=True, remove_none=True),
+        }
+
+    def run(
+        self,
+        elements_filepath: Path,
+        file_data: FileData,
+        output_dir: Path,
+        output_filename: str,
+        **kwargs: Any,
+    ) -> Path:
+        with open(elements_filepath) as elements_file:
+            elements_contents = json.load(elements_file)
+        for element in elements_contents:
+            self.conform_dict(data=element)
+        normalized_elements = []
+        for element in elements_contents:
+            normalized_elements.append(self.normalize_dict(data=element))
+
+        output_path = Path(output_dir) / Path(f"{output_filename}.json")
+        with open(output_path, "w") as output_file:
+            json.dump(normalized_elements, output_file)
+        return output_path
+
+
+@dataclass
+class ChromaUploaderConfig(UploaderConfig):
+    batch_size: int = 100
+
+
+@dataclass
+class ChromaUploader(Uploader):
+    upload_config: ChromaUploaderConfig
+    connection_config: ChromaConnectionConfig
+    _collection: Optional["ChromaCollection"] = None
+
+    def __post_init__(self):
+        self._collection = self.create_collection()
+
+    def is_async(self) -> bool:
+        return False
+
+    @requires_dependencies(["chromadb"], extras="chroma")
+    def create_collection(self) -> "ChromaCollection":
+        import chromadb
+
+        if self.connection_config.path:
+            chroma_client = chromadb.PersistentClient(
+                path=self.connection_config.path,
+                settings=self.connection_config.settings,
+                tenant=self.connection_config.tenant,
+                database=self.connection_config.database,
+            )
+
+        elif self.connection_config.host and self.connection_config.port:
+            chroma_client = chromadb.HttpClient(
+                host=self.connection_config.host,
+                port=self.connection_config.port,
+                ssl=self.connection_config.ssl,
+                headers=self.connection_config.access_config.headers,
+                settings=self.connection_config.access_config.settings,
+                tenant=self.connection_config.tenant,
+                database=self.connection_config.database,
+            )
+        else:
+            raise ValueError("Chroma connector requires either path or host and port to be set.")
+
+        collection = chroma_client.get_or_create_collection(
+            name=self.connection_config.collection_name
+        )
+        return collection
+
+    @DestinationConnectionError.wrap
+    def upsert_batch(self, batch):
```

Comment:
Might want to have this configurable if a user wants for an error to be thrown if it's a duplicate. Use upsert by default?

Reply:
A little over a month ago we decided that upsert was the preferred default behaviour. Passed this through Ryan and Matt. https://github.com/Unstructured-IO/unstructured/pull/2926